### PR TITLE
Update recipes to current standard and regenerate zproject

### PR DESCRIPTION
**Pull request by jimklimov**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -48,18 +50,29 @@ env:
...
 #   - BUILD_TYPE=check-py
 
-pkg_deps_prereqs: &pkg_deps_prereqs
+# Prerequisite packages provided by OS distro and used "as is"
+pkg_deps_prereqs_distro: &pkg_deps_prereqs_distro
+
+# Prerequisite packages that may be built from source or used from
+# prebuilt packages of that source (usually not from an OS distro)
+pkg_deps_prereqs_source: &pkg_deps_prereqs_source
     - libsodium-dev
```
Comment (aquette) [inline .travis.yml:64]: @jimklimov so you have uncommented the pkg prereqs?! I thought we were supposed to comment the ones pulled when repos is mentioned, no?

**Review by aquette — APPROVED** [review]
LGTM apart from my comment on pkg pre reqs

File: .travis.yml (hunk 1 above)
Comment (jimklimov) [reply .travis.yml:64]: In the new layout, the packages that can be built from git/tar source vs. those from the OS "as is" go to different YAML lists.

Below we comment away the source list's inclusion into the legacy-named list of orereqs in general. It is less error-prone (was harder to tell apart in same list what we build and what we don't - and to find that we missed a github reference to some log4cplus fork etc. ;) ) and easier to review changes in later regens (less differing lines).

Comment (jimklimov) [reply .travis.yml:64]: In some other components we did erroneously comment away some system-provided prereqs (libssl, libsasl); in others we did not comment away our forks (zmq stack, log4cplus) so tested against system/upstream packaged builds. Nothing exploded in travis for those (probably libssl came with ssh, etc.) but testing in different env than our production was uncool :\

Comment (aquette) [reply .travis.yml:64]: I indeed missed the _distro suffix... reviewing on a phone while bbq warms up 😁 thanks for the clarification

**Pull request merged**